core/node/node_framework/src/implementations/layers/web3_api/tx_sender.rs

```diff
@@ -0,0 +1,160 @@
+        // Get required resources.
+        let tx_sink = context.get_resource::<TxSinkResource>().await?.0;
+        let pool_resource = context.get_resource::<ReplicaPoolResource>().await?;
+        let replica_pool = pool_resource.get().await?;
+        let sealer = match context.get_resource::<ConditionalSealerResource>().await {
+            Ok(sealer) => Some(sealer.0.clone()),
+            Err(WiringError::ResourceLacking(_)) => None,
+            Err(other) => return Err(other),
+        };
+        let fee_input = context.get_resource::<FeeInputResource>().await?.0;
+
+        // Initialize Postgres caches.
+        let factory_deps_capacity = self.postgres_storage_caches_config.factory_deps_cache_size;
+        let initial_writes_capacity = self
+            .postgres_storage_caches_config
+            .initial_writes_cache_size;
+        let values_capacity = self.postgres_storage_caches_config.latest_values_cache_size;
+        let mut storage_caches =
+            PostgresStorageCaches::new(factory_deps_capacity, initial_writes_capacity);
+
+        if values_capacity > 0 {
+            let values_cache_task = storage_caches.configure_storage_values_cache(
+                values_capacity,
+                replica_pool.clone(),
+                context.runtime_handle().clone(),
+            );
+            context.add_task(Box::new(PostgresStorageCachesTask {
+                task: Box::new(values_cache_task),
+            }));
+        }
+
+        // Initialize `VmConcurrencyLimiter`.
+        let (vm_concurrency_limiter, vm_concurrency_barrier) =
+            VmConcurrencyLimiter::new(self.max_vm_concurrency);
+        context.add_task(Box::new(VmConcurrencyBarrierTask {
+            barrier: vm_concurrency_barrier,
+        }));
+
+        // Build `TxSender`.
+        let mut tx_sender = TxSenderBuilder::new(self.tx_sender_config, replica_pool, tx_sink);
+        if let Some(sealer) = sealer {
+            tx_sender = tx_sender.with_sealer(sealer);
+        }
+        let tx_sender = tx_sender
+            .build(
+                fee_input,
+                Arc::new(vm_concurrency_limiter),
+                self.api_contracts,
+                storage_caches,
+            )
+            .await;
+        context.insert_resource(TxSenderResource(tx_sender))?;
+
+        Ok(())
+    }
+}
+
+struct PostgresStorageCachesTask {
+    task: Box<dyn FnOnce() -> anyhow::Result<()> + Send>,
+}
+
+impl fmt::Debug for PostgresStorageCachesTask {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("PostgresStorageCachesTask")
+            .finish_non_exhaustive()
+    }
+}
+
+#[async_trait::async_trait]
+impl Task for PostgresStorageCachesTask {
+    fn name(&self) -> &'static str {
+        "postgres_storage_caches"
+    }
+
+    async fn run(self: Box<Self>, _stop_receiver: StopReceiver) -> anyhow::Result<()> {
+        tokio::task::spawn_blocking(self.task).await?
```

Comment:
Future work (?): May make sense to make this task cancel-aware and maybe have a nameable type.

Reply:
Makes sense, will tackle it in a separate PR.